packages/block-editor/src/components/rich-text/use-paste-handler.js

```diff
@@ -150,7 +150,7 @@ export function usePasteHandler( props ) {
 				return;
 			}
 
-			if ( pastePlainText ) {
+			if ( pastePlainText || ( ! files.length && ! html ) ) {
 				onChange( insert( value, create( { text: plainText } ) ) );
```

Comment:
👋🏻 It feels like we're patching for an edge case, it reads like: "if `pastePlainText ` OR any other option except those we handle". What if it is too generic and it hides future bugs :)? If it is plain text pasted what is preventing `pastePlainText ` to be enough?

Reply:
Thanks a lot @ellatrix. The context provided by you was very helpful. 
> It seems the entire text will be inserted into the rich text field, which doesn't seem right. If we paste
Yes, it was indeed the behaviour of the code. So after brainstorming a bit, I decided to implement my solution in a different way. 

Please check the solution and LMK how you feel about it.